### PR TITLE
created draft adr

### PR DESCRIPTION
**Pull request by McKnight-42**
First Draft of a ADR to get feedback upon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,4 +44,9 @@ Once your adapter is built out a few options.
 
 ## Code of Conduct
 
-Everyone interacting in the dbt project's codebases, issue trackers, chat rooms, and mailing lists is expected to follow the [dbt Code of Conduct](https://community.getdbt.com/code-of-conduct).
\ No newline at end of file
+Everyone interacting in the dbt project's codebases, issue trackers, chat rooms, and mailing lists is expected to follow the [dbt Code of Conduct](https://community.getdbt.com/code-of-conduct).
+
+
+<!-- "directory_name": "{{cookiecutter.adapter_name.lower().replace(' ','')}}",
```
Comment (leahwicz) [inline README.md:50]: What are these bottom lines here for?

File: docs/arch/adr-001-new-adapter-scaffolding.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## Context
+Currently we offer the ability for users and datawarehouses to build vendor/community supported adapter plugins via the [`create_adapter_plugins.py`](https://github.com/dbt-labs/dbt/blob/47033c459f2c835d81cc845d49ef23e7d19736b6/core/scripts/create_adapter_plugins.py) file. While this file is able to produce a usable adapter plugin, it is not the most user friendly experience as we have noted over time and needed to be updated.
+
+### known issues or possible improvements to be made
+- Updating script to generate a new adapter plugin is tricky, must edit numerous template strings in python code, difficult to test and search for all areas changes are needed. ex. `$ python create_adapter_plugins.py --sql --title-case=MyAdapter ./ myadapter`
+- Not an interactive experience, must know and pass all required arguments to the .py file 
+- stuck with defaults, or missing dbt suggested default dependencies
+  - Options to set up Github Actions
+  - flake8 by default for linting to match other adapters
+  - mypy by default or optional for stati type checking
```
Comment (leahwicz) [inline docs/arch/adr-001-new-adapter-scaffolding.md:12]:
```suggestion
  - mypy by default or optional for static type checking
```

File: docs/arch/adr-001-new-adapter-scaffolding.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - [`yehua`](https://yehua.readthedocs.io/en/latest/)
+        - Seems to of had similar functionality to cookiecutter but hasn't been suppoted in quite some time therefore no longer works with pip install, (unusable)
+
+## Decision
+After looking at available tooling, we have decided to use `cookiecutter` parsed through .py and generated "adapter" base file from current process and pulled out any information we would need to keep, built out the `cookiecutter.json` based variables that were built by running the .py variables and put them in way to augment things like adapter name to fit into correct place, assign correct ConnectionManager classes based on certain choices, applied file sturcture and dependency list for things dbt wants to suggest be used (ex. tox, flake8,etc.), once base file structure was built out I grabbed examples of the macros that do not have a default version and provided them as a context to be looked at, built out base steps of what all other macros do in general terms to help users make any changes to default action needed, created general docstrings for classes based on offical documentation and implementation.
```
Comment (leahwicz) [inline docs/arch/adr-001-new-adapter-scaffolding.md:36]: The wording in this paragraph is a little confusing. Could we try and make it a little clearer?

File: README.md (hunk 1 above)
Comment (McKnight-42) [reply README.md:50]: sorry those were hold overs from a different branch that should of gone away, removed

File: docs/arch/adr-001-new-adapter-scaffolding.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# New Adapter Scaffold
+
+## Context
+Currently we offer the ability for users and datawarehouses to build vendor/community supported adapter plugins via the [`create_adapter_plugins.py`](https://github.com/dbt-labs/dbt/blob/47033c459f2c835d81cc845d49ef23e7d19736b6/core/scripts/create_adapter_plugins.py) file. While this file is able to produce a usable adapter plugin, it is not the most user friendly experience as we have noted over time and needed to be updated.
+
+### known issues or possible improvements to be made
+- Updating script to generate a new adapter plugin is tricky, must edit numerous template strings in python code, difficult to test and search for all areas changes are needed. ex. `$ python create_adapter_plugins.py --sql --title-case=MyAdapter ./ myadapter`
```
[the remark this answers is not in the document]
Comment (McKnight-42) [reply docs/arch/adr-001-new-adapter-scaffolding.md:7]: have made suggested changes

File: docs/arch/adr-001-new-adapter-scaffolding.md (hunk 3 above)
Comment (McKnight-42) [reply docs/arch/adr-001-new-adapter-scaffolding.md:36]: rewrote section into two parts, one for choice of cookiecutter and build out of scaffold, one for addition of examples, and stubs, could eventually had a third for testing if we like it.

**Review by leahwicz — APPROVED** [review]

File: docs/arch/adr-001-new-adapter-scaffolding.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      - [`pyscaffold`](https://cookiecutter.readthedocs.io/en/1.7.2/overview.html)
+        - creates a simple scaffold of project not as flexible as cookiecutter
+      - [`yehua`](https://yehua.readthedocs.io/en/latest/)
+        - Seems to of had similar functionality to cookiecutter but hasn't been suppoted in quite some time therefore no longer works with pip install, (unusable)
+
+## Decision
+
+Firstly after looking at available tooling, we landed on using `cookiecutter` to aid in creating a interactive scaffolding session for users quickily generate a starting point to build out their adapter plugin by quickly building out variable names, giving choice selections for things like `adapter_src` which is how we let the program know which connection methods to pull in from `dbt-core`. This also meant we could keep much of the structure for the files the same from previous adapter generator by having the files exsit in the starting state and just passing the user passed parameters on generation.
```
Comment (kwigley) [inline docs/arch/adr-001-new-adapter-scaffolding.md:40]:
```suggestion
Firstly after looking at available tooling, we landed on using `cookiecutter` to aid in creating a interactive scaffolding session for users quickily generate a starting point to build out their adapter plugin by quickly building out variable names, giving choice selections for things like `adapter_src` which is how we let the program know which connection methods to pull in from `dbt-core`. This also meant we could keep much of the structure for the files the same from previous adapter generator by having the files exist in the starting state and just passing the user passed parameters on generation. 
```

**Review by kwigley — APPROVED** [review]

**Pull request merged**